clients/python/treedb_client/src/treedb_client/client.py

```
@@ -702,3 +771,19 @@ def _is_timeout(reason: Any) -> bool:
     if isinstance(reason, (socket.timeout, TimeoutError)):
         return True
     return "timed out" in str(reason).lower()
+
+
+def _is_broken_connection(reason: Any) -> bool:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `reason`

(ANN401)